services/src/datasets/in_memory.rs

```diff
@@ -226,18 +235,17 @@ impl MetaDataProvider<GdalLoadingInfo, RasterResultDescriptor> for HashMapDataSe
         Box<dyn MetaData<GdalLoadingInfo, RasterResultDescriptor>>,
         geoengine_operators::error::Error,
     > {
-        Ok(Box::new(
-            self.gdal_data_sets
-                .get(&data_set.internal().ok_or(
-                    geoengine_operators::error::Error::DataSetMetaData {
-                        source: Box::new(error::Error::DataSetIdTypeMissMatch),
-                    },
-                )?)
-                .ok_or(geoengine_operators::error::Error::DataSetMetaData {
-                    source: Box::new(error::Error::UnknownDataSetId),
-                })?
-                .clone(),
-        ))
+        Ok(self
+            .gdal_data_sets
+            .get(&data_set.internal().ok_or(
+                geoengine_operators::error::Error::DataSetMetaData {
+                    source: Box::new(error::Error::DataSetIdTypeMissMatch),
+                },
+            )?)
+            .ok_or(geoengine_operators::error::Error::DataSetMetaData {
+                source: Box::new(error::Error::UnknownDataSetId),
+            })?
+            .clone())
```

Comment:
This is super hard to read, maybe you can just use a match block?

Reply:
on what should I match? I can split it up a bit if that helps:

```
        let id = data_set
            .internal()
            .ok_or(geoengine_operators::error::Error::DataSetMetaData {
                source: Box::new(error::Error::DataSetIdTypeMissMatch),
            })?;

        Ok(self
            .gdal_data_sets
            .get(&id)
            .ok_or(geoengine_operators::error::Error::DataSetMetaData {
                source: Box::new(error::Error::UnknownDataSetId),
            })?
            .clone())
```

Reply:
d1fc0c0f3586c3aee2d72918ee459b3fd25759ec